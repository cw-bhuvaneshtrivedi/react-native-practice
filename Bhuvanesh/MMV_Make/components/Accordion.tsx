import { Text, TouchableOpacity, StyleSheet, View } from "react-native";
import React, { ForwardedRef, useEffect, useState } from "react";
import { Entypo } from "@expo/vector-icons";
import AccordionAnimated from "./AccordionAnimated";
import Animated, {
  useAnimatedStyle,
  useSharedValue,
  withTiming,
} from "react-native-reanimated";

type AccordionProps = {
  makeName: string;
  version: string[];
  setView: (index: number) => { index: number };
  index: number;
  view: number;
};
const Accordion = ({
  makeName,
  version,
  setView,
  index,
  view,
}: AccordionProps) => {
  const [open, setOpen] = useState(false);
  const angle = useSharedValue(0);

  const handlePress = () => {
    setOpen(!open);
    if (open) {
      angle.value = withTiming(0);
      setView(-1);
    } else {
      angle.value = withTiming(180);
      // setOpen(false);
      setView(index);
    }
  };

  useEffect(() => {
    if (view != index && open) {
      setOpen(false);
      angle.value = withTiming(0);
    }
  }, [view]);

  const rotateIcon = useAnimatedStyle(() => ({
    transform: [{ rotateZ: `${angle.value}deg` }],
  }));

  return (
    <View style={styles.container}>
      <TouchableOpacity
        style={styles.flexRow}
        onPress={handlePress}
        testID="drawer"
      >
        <Text style={styles.text}>{makeName}</Text>
        <Animated.View style={rotateIcon}>
          <Entypo name="chevron-down" size={24} color="#aaaaaa" />
        </Animated.View>
      </TouchableOpacity>
      <AccordionAnimated open={open} data={version} />
      <View style={styles.border}></View>
    </View>
  );
};
export default Accordion;
const styles = StyleSheet.create({
  container: {
    overflow: "hidden",
  },
  flexRow: {
    paddingLeft: 28,
    height: 54,
    paddingRight: 28,
    flexDirection: "row",
    justifyContent: "space-between",
    alignItems: "center",
  },
  text: {
    fontSize: 15,
  },
  border: {
    height: 1,
    backgroundColor: "#efefef",
  },
});
