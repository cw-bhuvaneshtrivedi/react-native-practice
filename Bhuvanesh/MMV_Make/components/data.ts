const data = [
  [
    "Item aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa",
    "Item aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa",
    "Item aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa",
  ],
  ["Item 1 1", "Item 1 2", "Item 1 3"],
];

export default data;
