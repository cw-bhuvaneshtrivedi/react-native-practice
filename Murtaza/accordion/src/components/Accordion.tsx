import { StyleSheet, Text, TouchableOpacity, View, Dimensions } from "react-native";
import React, { useState, useEffect } from "react";
import { Entypo } from "@expo/vector-icons";
import AccordionItem from "./AccordionItem";
import Animated, {
  useAnimatedStyle,
  useSharedValue,
  withTiming,
  Easing,
} from "react-native-reanimated";

interface AccordionProps {
  title: string;
  items: string[];
  isOpen: boolean;
  onPress: () => void;
}

const Accordion: React.FC<AccordionProps> = ({ title, items, isOpen, onPress }) => {
  const rotateZ = useSharedValue(isOpen ? -180 : 0);

  useEffect(() => {
    rotateZ.value = withTiming(isOpen ? -180 : 0, {
      duration: 350,
      easing: Easing.inOut(Easing.quad),
    });
  }, [isOpen]);

  const handleAccordionPress = () => {
    onPress();
  };

  const animatedStyle = useAnimatedStyle(() => ({
    transform: [{ rotateZ: `${rotateZ.value}` + "deg" }],
  }));

  return (
    <View>
      <TouchableOpacity style={styles.container} onPress={handleAccordionPress}>
        <Text style={styles.accordionTitle}>{title}</Text>
        <Animated.View style={animatedStyle}>
          <Entypo name="chevron-down" size={24} color="black" />
        </Animated.View>
      </TouchableOpacity>
      <View style={styles.line}></View>

      {isOpen && (
        <View>
          {items.map((item, index) => (
            <AccordionItem key={index} itemName={item} />
          ))}
        </View>
      )}
    </View>
  );
};

export default Accordion;

const styles = StyleSheet.create({
  container: {
    backgroundColor: "white",
    paddingTop: 16,
    paddingLeft: 28,
    paddingRight: 31,
    flexDirection: "row",
    justifyContent: "space-between",
    height: 50,
    // marginTop: 20,
  },
  accordionTitle: {
    color: "#484848",
    fontSize: 15,
  },
  line: {
    backgroundColor: "#efefef",
    width: "100%",
    height: 1,
  },
});
