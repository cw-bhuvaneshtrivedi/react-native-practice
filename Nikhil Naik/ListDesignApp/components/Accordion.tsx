import {
  ScrollView,
  StyleSheet,
  Text,
  TouchableOpacity,
  View,
} from "react-native";
import React, { useEffect, useState } from "react";
import { Feather } from "@expo/vector-icons";
import AccordionAnimation from "./AccordionAnimation";
import Animated, {
  useAnimatedStyle,
  useSharedValue,
  withTiming,
} from "react-native-reanimated";

const Accordion = ({ data, idx, setView, view }) => {
  const [close, setClose] = useState(false);
  const rotateZ = useSharedValue(0);
  const style = useAnimatedStyle(() => ({
    transform: [{ rotateZ: `${rotateZ.value}` + "deg" }],
  }));
  const handle = () => {
    if (rotateZ.value == -180)
      rotateZ.value = withTiming(rotateZ.value + 180, { duration: 300 });
    else rotateZ.value = withTiming(rotateZ.value - 180, { duration: 300 });
  };
  useEffect(() => {
    if (close && view != -1 && view != idx) {
      handle();
      setClose(false);
    }
  }, [view]);
  const handlePress = () => {
    if (close) {
      setClose(!close);
      handle();
    } else {
      setClose(!close);
      handle();
      setView((item: number) => {
        return idx;
      });
    }
  };
  return (
    <ScrollView>
      <TouchableOpacity onPress={handlePress} testID="accordionButton">
        <View style={styles.container}>
          <Text style={styles.accordionText}>{data.makeName}</Text>
          <Animated.View style={[style]}>
            <Feather name="chevron-down" size={30} color="#aaaaaa" />
          </Animated.View>
        </View>
      </TouchableOpacity>
      <View style={styles.seperator}></View>
      <AccordionAnimation open={close} data={data} />
    </ScrollView>
  );
};

export default Accordion;

const styles = StyleSheet.create({
  container: {
    paddingLeft: 28,
    paddingRight: 30,
    height: 50,
    flexDirection: "row",
    justifyContent: "space-between",
    alignItems: "center",
  },
  accordionText: {
    fontSize: 15,
    color: "#484848",
  },
  seperator: {
    width: "100%",
    height: 2,
    backgroundColor: "#efefef",
  },
});
