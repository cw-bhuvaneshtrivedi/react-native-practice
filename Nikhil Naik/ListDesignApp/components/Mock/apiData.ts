export const apiData = [
  {
    modelId: 1729,
    modelName: "Creta",
    maskingName: "creta",
    rootId: "382",
    makeId: 8,
    makeName: "Hyundai",
    makeMaskingName: "hyundai",
    launchedOn: "01/16/2024 00:00:00",
    discontinuationDate: "",
    rootName: "Creta",
    status: "New",
    imagePath:
      "/n/cw/ec/106815/creta-exterior-right-front-three-quarter-4.png?isig=0",
    versionId: 9371,
  },
  {
    modelId: 1729,
    modelName: "Creta",
    maskingName: "creta",
    rootId: "382",
    makeId: 8,
    makeName: "Hyundai",
    makeMaskingName: "hyundai",
    launchedOn: "01/16/2024 00:00:00",
    discontinuationDate: "",
    rootName: "Creta",
    status: "New",
    imagePath:
      "/n/cw/ec/106815/creta-exterior-right-front-three-quarter-4.png?isig=0",
    versionId: 9371,
  },
  {
    modelId: 1729,
    modelName: "Ecco",
    maskingName: "Ecco",
    rootId: "382",
    makeId: 9,
    makeName: "Maruti Suzuki",
    makeMaskingName: "Maruti Suzuki",
    launchedOn: "01/16/2024 00:00:00",
    discontinuationDate: "",
    rootName: "Ecco",
    status: "New",
    imagePath:
      "/n/cw/ec/106815/creta-exterior-right-front-three-quarter-4.png?isig=0",
    versionId: 9371,
  },
  {
    modelId: 1729,
    modelName: "Ecco",
    maskingName: "Ecco",
    rootId: "382",
    makeId: 9,
    makeName: "Maruti Suzuki",
    makeMaskingName: "Maruti Suzuki",
    launchedOn: "01/16/2024 00:00:00",
    discontinuationDate: "",
    rootName: "Ecco",
    status: "New",
    imagePath:
      "/n/cw/ec/106815/creta-exterior-right-front-three-quarter-4.png?isig=0",
    versionId: 9371,
  },
];
