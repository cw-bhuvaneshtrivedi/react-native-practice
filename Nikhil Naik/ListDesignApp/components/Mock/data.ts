export const data = [
  [
    "Nikhil",
    "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Vestibulum lobortis tellus dolor. Donec sed enim fringilla, congue tortor at, venenatis nunc. Morbi in magna vitae ex iaculis vehicula non non sem. Sed euismod in nulla eget feugiat. Praesent tempor molestie leo, id iaculis lectus. Pellentesque a nulla nunc. Aliquam id.",
    "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla mi.",
  ],
  [
    "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Vestibulum lobortis tellus dolor. Donec sed enim fringilla, congue tortor at, venenatis nunc. Morbi in magna vitae ex iaculis vehicula non non sem. Sed euismod in nulla eget feugiat. Praesent tempor molestie leo, id iaculis lectus. Pellentesque a nulla nunc. Aliquam id.",
    "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla mi.",
  ],
  [
    "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla miLorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla mi",
    "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Vestibulum lobortis tellus dolor. Donec sed enim fringilla, congue tortor at, venenatis nunc. Morbi in magna vitae ex iaculis vehicula non non sem. Sed euismod in nulla eget feugiat. Praesent tempor molestie leo, id iaculis lectus. Pellentesque a nulla nunc. Aliquam id.",
    "Lorem ipsum dolor sit amet",
  ],
  [
    "Nikhil",
    "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Vestibulum lobortis tellus dolor. Donec sed enim fringilla, congue tortor at, venenatis nunc. Morbi in magna vitae ex iaculis vehicula non non sem. Sed euismod in nulla eget feugiat. Praesent tempor molestie leo, id iaculis lectus. Pellentesque a nulla nunc. Aliquam id.",
  ],
  ["Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla mi."],
];
