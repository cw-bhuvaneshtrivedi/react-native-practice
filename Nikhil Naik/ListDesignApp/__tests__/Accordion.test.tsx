import Accordion from "../components/Accordion";
import {
  fireEvent,
  render,
  screen,
  userEvent,
} from "@testing-library/react-native";
import { data } from "../components/Mock/data";
import React from "react";

const setClose = jest.fn();
const setView = jest.fn();

describe("Accordion ", () => {
  beforeEach(() => {
    setClose.mockImplementation((open) => {
      return !open;
    });
    setView.mockImplementation((num) => num);
    render(<Accordion data={data[0]} idx={0} setView={setView} view={0} />);
    setClose(false);
  });

  it("should have button", () => {
    const result = screen.getAllByTestId("accordionButton");
    expect(result).toHaveLength(1);
  });

  it("should have responsive button", async () => {
    const button = await screen.findByTestId("accordionButton");
    fireEvent(button, "click");
    expect(setClose).toHaveReturnedWith(true);
    fireEvent(button, "click");
    expect(setClose).toHaveBeenCalledWith(false);
    fireEvent(button, "click");
    expect(setClose).toHaveReturnedWith(true);
    expect(setView).toHaveBeenCalledTimes(2);
  });
});
