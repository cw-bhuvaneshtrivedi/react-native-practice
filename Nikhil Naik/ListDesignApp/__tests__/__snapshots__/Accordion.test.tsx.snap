// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion  should have responsive button 1`] = `
<RCTScrollView>
  <View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
      testID="accordionButton"
    >
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "height": 50,
            "justifyContent": "space-between",
            "paddingLeft": 28,
            "paddingRight": 30,
          }
        }
      >
        <Text
          style={
            {
              "color": "#484848",
              "fontSize": 15,
            }
          }
        >
          Maruti
        </Text>
        <View
          animatedStyle={
            {
              "value": {
                "transform": [
                  {
                    "rotateZ": "-180deg",
                  },
                ],
              },
            }
          }
          collapsable={false}
          style={
            {
              "transform": [
                {
                  "rotateZ": "0deg",
                },
              ],
            }
          }
        >
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              [
                {
                  "color": "#aaaaaa",
                  "fontSize": 30,
                },
                undefined,
                {
                  "fontFamily": "feather",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                {},
              ]
            }
          >
            ï„®
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        {
          "backgroundColor": "#efefef",
          "height": 2,
          "width": "100%",
        }
      }
    />
    <View
      animatedStyle={
        {
          "value": {
            "height": 0,
          },
        }
      }
      collapsable={false}
      style={
        {
          "height": 0,
          "overflow": "hidden",
        }
      }
      testID="card"
    >
      <View
        style={
          {
            "position": "absolute",
          }
        }
      >
        <RCTScrollView
          data={
            [
              "Nikhil",
              "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Vestibulum lobortis tellus dolor. Donec sed enim fringilla, congue tortor at, venenatis nunc. Morbi in magna vitae ex iaculis vehicula non non sem. Sed euismod in nulla eget feugiat. Praesent tempor molestie leo, id iaculis lectus. Pellentesque a nulla nunc. Aliquam id.",
              "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla mi.",
            ]
          }
          getItem={[Function]}
          getItemCount={[Function]}
          keyExtractor={[Function]}
          onContentSizeChange={[Function]}
          onLayout={[Function]}
          onMomentumScrollBegin={[Function]}
          onMomentumScrollEnd={[Function]}
          onScroll={[Function]}
          onScrollBeginDrag={[Function]}
          onScrollEndDrag={[Function]}
          removeClippedSubviews={false}
          renderItem={[Function]}
          scrollEventThrottle={0.0001}
          stickyHeaderIndices={[]}
          viewabilityConfigCallbackPairs={[]}
        >
          <View>
            <View
              onFocusCapture={[Function]}
              onLayout={[Function]}
              style={null}
            >
              <View>
                <View
                  style={
                    {
                      "alignItems": "center",
                      "backgroundColor": "#f9f9f9",
                      "flexDirection": "row",
                      "justifyContent": "space-between",
                      "paddingLeft": 36,
                      "paddingRight": 30,
                    }
                  }
                >
                  <Text
                    style={
                      {
                        "color": "#484848",
                        "fontSize": 15,
                        "paddingBottom": 16,
                        "paddingTop": 16,
                      }
                    }
                    testID="cardName"
                  >
                    Nikhil
                  </Text>
                </View>
                <View
                  style={
                    {
                      "backgroundColor": "#efefef",
                      "height": 2,
                      "width": "100%",
                    }
                  }
                />
              </View>
            </View>
            <View
              onFocusCapture={[Function]}
              onLayout={[Function]}
              style={null}
            >
              <View>
                <View
                  style={
                    {
                      "alignItems": "center",
                      "backgroundColor": "#f9f9f9",
                      "flexDirection": "row",
                      "justifyContent": "space-between",
                      "paddingLeft": 36,
                      "paddingRight": 30,
                    }
                  }
                >
                  <Text
                    style={
                      {
                        "color": "#484848",
                        "fontSize": 15,
                        "paddingBottom": 16,
                        "paddingTop": 16,
                      }
                    }
                    testID="cardName"
                  >
                    Lorem ipsum dolor sit amet, consectetur adipiscing elit. Vestibulum lobortis tellus dolor. Donec sed enim fringilla, congue tortor at, venenatis nunc. Morbi in magna vitae ex iaculis vehicula non non sem. Sed euismod in nulla eget feugiat. Praesent tempor molestie leo, id iaculis lectus. Pellentesque a nulla nunc. Aliquam id.
                  </Text>
                </View>
                <View
                  style={
                    {
                      "backgroundColor": "#efefef",
                      "height": 2,
                      "width": "100%",
                    }
                  }
                />
              </View>
            </View>
            <View
              onFocusCapture={[Function]}
              onLayout={[Function]}
              style={null}
            >
              <View>
                <View
                  style={
                    {
                      "alignItems": "center",
                      "backgroundColor": "#f9f9f9",
                      "flexDirection": "row",
                      "justifyContent": "space-between",
                      "paddingLeft": 36,
                      "paddingRight": 30,
                    }
                  }
                >
                  <Text
                    style={
                      {
                        "color": "#484848",
                        "fontSize": 15,
                        "paddingBottom": 16,
                        "paddingTop": 16,
                      }
                    }
                    testID="cardName"
                  >
                    Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla mi.
                  </Text>
                </View>
                <View
                  style={
                    {
                      "backgroundColor": "#efefef",
                      "height": 2,
                      "width": "100%",
                    }
                  }
                />
              </View>
            </View>
          </View>
        </RCTScrollView>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
